portfolio/src/main/webapp/script.js

```diff
@@ -11,18 +11,68 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
- 
+
+var userLoggedIn;
+
+function getLink(){
+    console.log("running getLink()...");
+    fetch("/log").then(response => response.json()).then((text) => {
+
+      console.log(text);
+      const jsonValues = Object.values(text);
+
+      var logUrl = jsonValues[0];
+      userLoggedIn = jsonValues[1];
+      
+      var loginLink = document.getElementById("loginLink");
+      loginLink.href = logUrl;
+
+      var loginStatus = document.getElementById("loginDescription");
+
+      if(userLoggedIn == "true"){ loginStatus.innerHTML = "LOG OUT HERE";}
+      else{ loginStatus.innerHTML = "LOG IN HERE";} 
+    });
+}
+
+
 function getComments(){
+  console.log("running getComments()...");
+  getLink();
   fetch('/data').then(response => response.json()).then((text) => {
     const commentContainer = document.getElementById('commentList');
     const jsonValues = Object.values(text);
- 
+    const jsonKeys = Object.keys(text);
+
     console.log(jsonValues);
-    
-    for(var i = 0; i < Object.values(text).length; i++){
-      var comment = document.createElement('p');
-      comment.innerHTML = jsonValues[i];
-      commentContainer.appendChild(comment);
-    }
-  })
+    console.log("User is logged in: " + userLoggedIn);
```

Comment:
these sorts of statements are typically for debugging purposes, you probably shouldn't include them in checked-in code =)